lib/src/working_copy.rs

```diff
@@ -51,9 +51,14 @@ use crate::repo_path::{FsPathParseError, RepoPath, RepoPathComponent, RepoPathJo
 use crate::store::Store;
 use crate::tree::{Diff, Tree};
 
+#[cfg(unix)]
```

Comment:
You are probably aware, but e.g. `wasm` won't be covered in either of these cases. (It looks like `wasm` currently doesn't support file permissions, as per the lack of `PermissionsExt` at https://doc.rust-lang.org/stable/std/os/wasi/fs/index.html).

Reply:
I didn't know that. I'll leave that for someone else to fix later.